ksqldb-streams/src/main/java/io/confluent/ksql/execution/streams/SourceBuilderUtils.java

```diff
@@ -366,35 +366,38 @@ public GenericRow transform(final K key, final GenericRow row) {
             return row;
           }
 
+          final GenericRow rowCopy = GenericRow.fromList(row.values());
```

Comment:
why are we copying the row? this might actually result in a non-trivial performance penalty since this is done per-row

Reply:
This was a bug in ksql that I caught actually through testing :) We should have been copying the row as Streams mutates the row when processing. It just so happened that the issue came up with self-joins but this is irrespective of the optimation. What happens is that the `transformValues` appends the pseudocolumns to the row but it mutates the original row. So, if the same row gets processed again, as in the case with the self-join, the pseudocolumns are appended again to the row. Twice which changes the schema and all following operators fail because of serde, etc.